src/expr.rs

```diff
@@ -139,7 +139,14 @@ pub(crate) fn format_expr(
         | ast::ExprKind::While(..) => to_control_flow(expr, expr_type)
             .and_then(|control_flow| control_flow.rewrite(context, shape)),
         ast::ExprKind::ConstBlock(ref anon_const) => {
-            Some(format!("const {}", anon_const.rewrite(context, shape)?))
+            if let ast::ExprKind::Block(ref block, label) = anon_const.value.kind {
+                Some(format!(
+                    "const {}",
+                    rewrite_block(block, Some(expr.attrs()), label, context, shape)?
+                ))
+            } else {
+                Some(format!("const {}", anon_const.rewrite(context, shape)?))
+            }
```

Comment:
Instead of duplicating the `Some(format!("const {}"))` code in both branches, let's pull that out and save it for the end. I think a nice way to rewrite this would be something like this:

```rust
            let rewrite = match anon_const.value.kind {
                ast::ExprKind::Block(ref block, opt_label) => {
                    // Inner attributes are associated with the `ast::ExprKind::ConstBlock` node,
                    // not the `ast::Block` node we're about to rewrite. To prevent dropping inner
                    // attributes call `rewrite_block` directly.
                    // See https://github.com/rust-lang/rustfmt/issues/6158
                    rewrite_block(block, Some(&expr.attrs), opt_label, context, shape)?
                }
                _ => anon_const.rewrite(context, shape)?,
            };
            Some(format!("const {}", rewrite))
```